Sprout/View Controllers/LoginViewController.m

```diff
@@ -0,0 +1,115 @@
+//
+//  LoginViewController.m
+//  Sprout
+//
+//  Created by Sarah Wang on 7/12/21.
+//
+
+#import "LoginViewController.h"
+#import "Parse/Parse.h"
+
+@interface LoginViewController ()
+
+@property (weak, nonatomic) IBOutlet UITextField *usernameField;
+@property (weak, nonatomic) IBOutlet UITextField *passwordField;
+@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *activityIndicator;
+@property (strong, nonatomic) UIAlertController *blankAlert;
+@property (strong, nonatomic) UIAlertController *registrationAlert;
+@property (strong, nonatomic) UIAlertController *loginAlert;
+
+@end
+
+@implementation LoginViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.blankAlert = [UIAlertController alertControllerWithTitle:@"Username or password is empty." message:@"Please try again!" preferredStyle:(UIAlertControllerStyleAlert)];
+    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+        // handle response here.
+    }];
+    [self.blankAlert addAction:okAction];
+    
+    self.registrationAlert = [UIAlertController alertControllerWithTitle:@"Error registering user." message:@"Please try again!" preferredStyle:(UIAlertControllerStyleAlert)];
+    [self.registrationAlert addAction:okAction];
+    
+    self.loginAlert = [UIAlertController alertControllerWithTitle:@"Error during user login." message:@"Please try again!" preferredStyle:(UIAlertControllerStyleAlert)];
+    [self.loginAlert addAction:okAction];
+
+}
+
+- (IBAction)didTap:(id)sender {
```

Comment:
This is interesting. Almost every IBAction is a tap... what is the actual purpose of this... `cancel` ?